pkg/apis/config/features_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2020 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	. "knative.dev/pkg/configmap/testing"
+	_ "knative.dev/pkg/system/testing"
+)
+
+func TestFeaturesConfigurationFromFile(t *testing.T) {
+	cm, example := ConfigMapsFromTestFile(t, FeaturesConfigName)
+
+	if _, err := NewFeaturesConfigFromConfigMap(cm); err != nil {
+		t.Errorf("NewDefaultsConfigFromConfigMap(actual) = %v", err)
+	}
+
+	if got, err := NewFeaturesConfigFromConfigMap(example); err != nil {
+		t.Errorf("NewDefaultsConfigFromConfigMap(example) = %v", err)
+	} else {
+		// Those are in example, to show usage,
+		// but default is nil, i.e. inheriting k8s.
+		// So for this test we ignore those, but verify the other fields.
+		got.EnableMultiContainer = false
+		if want := defaultFeaturesConfig(); !cmp.Equal(got, want) {
+			t.Errorf("Example does not represent default config: diff(-want,+got)\n%s",
+				cmp.Diff(want, got))
```

Comment:
While you're at it :)
```suggestion
			t.Error("Example does not represent default config: diff(-want,+got)\n",
				cmp.Diff(want, got))
```

Reply:
That's not equivalent, `t.Error` adds a space between arguments, this would make the first line of the diff misaligned. I changed the `Errorf` above though.